### ISSUE TITLE
bug: Page transitions don't work with custom header components

### ISSUE THREAD

**Issue by moldstadt** [ionic-team/ionic-framework#21808]
# Bug Report

**Ionic version:**

[ ] **4.x**
[x] **5.x**

**Current behavior:**

When transitioning between pages on iOS, the headers and toolbars don't animate if they are in a custom component.

**Expected behavior:**

Page transitions shouldn't be affected by wrapping headers or toolbars in custom components.

**Steps to reproduce:**

Create a custom component that wraps a header or toolbar and use it instead of ion-header or ion-toolbar. 
Use this component on a page.

**Related code:**
Custom component template:
```
<ion-header>
    <ion-toolbar>
        <ion-title>My Header</ion-title>
    </ion-toolbar>
</ion-header>
```

**Other information:**

The problem is related to the selectors in ios.transition.ts, which seem to be too specific. For example:
`const headerEls = enteringEl.querySelectorAll(':scope > ion-header > *:not(ion-toolbar), :scope > ion-footer > *');`
By changing the child selectors to be descendants, solves the problem.
`const headerEls = enteringEl.querySelectorAll(':scope ion-header > *:not(ion-toolbar), :scope ion-footer > *');`

There are several places where this change needs to be made in ios.transitions.md. I'm not sure if these changes will have any undesirable side effects.

**Ionic info:**

```
Ionic:

   Ionic CLI                     : 5.4.15 (C:\Users\Mike\AppData\Roaming\npm\node_modules\ionic)
   Ionic Framework               : @ionic/angular 5.3.0
   @angular-devkit/build-angular : 0.901.12
   @angular-devkit/schematics    : 9.1.12
   @angular/cli                  : 9.1.12
   @ionic/angular-toolkit        : 2.3.0

Capacitor:

   Capacitor CLI   : 1.3.0
   @capacitor/core : 1.3.0

Utility:

   cordova-res (update available: 0.9.0) : 0.8.1
   native-run                            : 0.3.0

System:

   NodeJS : v10.15.3 (C:\Program Files\nodejs\node.exe)
   npm    : 6.4.1
   OS     : Windows 10
```

**Comment by liamdebeasi** [comment]
Thanks for the issue. I am going to close this as a duplicate of https://github.com/ionic-team/ionic-framework/issues/18130#issuecomment-643333628. The reason why this is not working is due to our use of `:scope` in the transition code as you have noted. We are not going to support this at this time as it will likely cause issues with nested `ion-nav` components.

**Comment by moldstadt** [comment]
Thanks @liamdebeasi. 

I had a look at your comment. The approach I took was slightly different, and I don't think it will affect nested ion-nav components. I simply changed the child selectors for the `ion-headers`, `ion-footers` and `ion-toolbars` to descendant selectors. This worked without removing the `:scope` selectors.

E.g., `const headerEls = enteringEl.querySelectorAll(':scope > ion-header > *:not(ion-toolbar), :scope > ion-footer > *');` becomes `const headerEls = enteringEl.querySelectorAll(':scope ion-header > *:not(ion-toolbar), :scope ion-footer > *');`

If this approach doesn't have any negative side-effects, would it not be worth fixing?
